core/src/main/java/io/questdb/std/MmapCache.java

```diff
@@ -0,0 +1,276 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.std;
+
+/**
+ * Thread-safe cache for memory-mapped file regions with reference counting.
+ * Reuses existing mappings for the same file when possible to reduce system calls.
+ */
+public class MmapCache {
+    private final LongObjHashMap<MmapCacheRecord> mmapAddrCache = new LongObjHashMap<>();
+    private final LongObjHashMap<MmapCacheRecord> mmapFileCache = new LongObjHashMap<>();
+    private long mmapReuseCount = 0;
+
+    /**
+     * Maps file region into memory, reusing existing mapping if available.
+     */
+    public long cacheMmap(int fd, long fileCacheKey, long len, long offset, int flags, int memoryTag) {
```

Comment:
we should have arg names aligned, it `fileCacheKey` is the `fd`, it should be called that